src/Language/Haskell/LSP/Test/Session.hs

```diff
@@ -220,8 +221,10 @@ runSessionWithHandles serverIn serverOut serverProc serverHandler config caps ro
   mainThreadId <- myThreadId
 
   let context = SessionContext serverIn absRootDir messageChan reqMap initRsp config caps
-      initState = SessionState (IdInt 0) mempty mempty 0 False Nothing
-      runSession' = runSession context initState
+      initState tmp_dir = SessionState (IdInt 0) (VFS mempty tmp_dir)
+                                       mempty 0 False Nothing
+      runSession' ses = withSystemTempDirectory "lsp-test" $ \tmp_dir ->
```

Comment:
I added a function to `haskell-lsp` to avoid having to manually call `withSystemTempDirectory`.

Reply:
which is called?

Reply:
`initVFS` - https://github.com/alanz/haskell-lsp/blob/master/src/Language/Haskell/LSP/VFS.hs#L78